packages/ingest/test/test-recursion.js

```diff
@@ -4,113 +4,175 @@ const test = require('ava');
 
 const recursion = require('../recursion');
 
-test('recursion lists all files in root and subdirectories when originalPath is empty', async (t) => {
-  const dirs = {
-    '': [
-      { type: 'd', name: 'dir' },
+test('recursion lists all files in default directory and subdirectories when originalPath is empty',
+  async (t) => {
+    const dirs = {
+      '.': [
+        { type: 'd', name: 'dir' },
+        { type: '-', name: 'file1' },
+        { type: '-', name: 'file2' }
+      ],
+      './dir': [
+        { type: '-', name: 'file3' }
+      ]
+    };
+    const fn = (path) => dirs[path];
+    const files = await recursion(fn, '');
+    t.deepEqual(files, [
+      { type: '-', name: 'file3' },
       { type: '-', name: 'file1' },
       { type: '-', name: 'file2' }
+    ]);
+  });
+
+test('recursion lists all files in a simple relative text path', async (t) => {
+  const dirs = {
+    '.': [
+      { type: 'd', name: 'path' }
     ],
-    dir: [
+    './path': [
+      { type: '-', name: 'file1' },
+      { type: '-', name: 'file2' },
       { type: '-', name: 'file3' }
     ]
   };
   const fn = (path) => dirs[path];
-  const files = await recursion(fn, '');
-  t.deepEqual(files, [
-    { type: '-', name: 'file3' },
-    { type: '-', name: 'file1' },
-    { type: '-', name: 'file2' }
-  ]);
+  const files = await recursion(fn, 'path');
+  t.deepEqual(files, dirs['./path']);
 });
 
-test('recursion lists all files in a simple directory structure with text path', async (t) => {
+test('recursion lists all files in a simple absolute text path', async (t) => {
   const dirs = {
-    '': [
+    '/': [
       { type: 'd', name: 'path' }
     ],
-    path: [
+    '/path': [
       { type: '-', name: 'file1' },
       { type: '-', name: 'file2' },
       { type: '-', name: 'file3' }
     ]
   };
   const fn = (path) => dirs[path];
-  const files = await recursion(fn, 'path');
-  t.deepEqual(files, dirs.path);
+  const files = await recursion(fn, '/path');
+  t.deepEqual(files, dirs['/path']);
 });
 
-test('recursion lists all files in a simple directory structure with regex path', async (t) => {
+test('recursion lists all files in a simple relative regex path', async (t) => {
   const dirs = {
-    '': [
+    '.': [
```

Comment:
Might be a bad idea, but do we need/want to include files at the starting directory in this test fixture to ensure that those **aren't** picked up as well?

Reply:
f92ab5faa